pilot/pkg/xds/ads.go

```diff
@@ -627,28 +627,16 @@ func (s *DiscoveryServer) removeCon(conID string) {
 
 // nolint
 func (conn *Connection) NonceAcked(typeUrl string) string {
-	wr := conn.proxy.GetWatchedResource(typeUrl)
-	if wr != nil {
-		return wr.NonceAcked
```

Comment:
`conn.proxy.GetWatchedResource` is protected by a lock. Here, we are just reading a string, so why is there a data race?

Reply:
But wr.NonceAcked is not locked. 

Take a look at the original issue

Reply:
Thanks, lgtm

Reply:
I see other places calling `GetWatchedResource` and reading its return value (similar to the `NonceAcked` function), typically RDS/EDS reading `WatchedResource.ResourceNames`, which seems to have similar problems?